deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -169,6 +193,55 @@ public boolean authenticate(AuthConfig authConfig) {
         // TODO
         return false;
     }
+    
+    
+	@Override
+	public int exec(K8sAuthorisation k8sAuthorisation, String servicename, String appname, String namespace,
+			String replicaName) throws HyscaleException {
+
+		if (k8sAuthorisation == null) {
+			throw new HyscaleException(DeployerErrorCodes.K8SAUTHORISATION_NOT_FOUND);
+		}
+		List<String> commands = null;
+		KubeConfig kubeConfig = null;
+		int exitCode = 0;
+
+		switch (k8sAuthorisation.getK8sAuthType()) {
+
+		case BASIC_AUTH:
+			K8sBasicAuth k8sBasicAuth = (K8sBasicAuth) k8sAuthorisation;
+			commands = deployCommandProvider.getExecCommandByBasicAuth(replicaName, namespace, k8sBasicAuth);
+			break;
+
+		case KUBE_CONFIG_FILE:
+			K8sConfigFileAuth k8sConfigFileAuth = (K8sConfigFileAuth) k8sAuthorisation;
+			kubeConfig = getKubeConfig(k8sConfigFileAuth);
+			if (kubeConfig.getAccessToken() != null) {
+				commands = deployCommandProvider.getExecCommandByAccessToken(replicaName, namespace,
+						kubeConfig.getAccessToken());
+				exitCode = CommandExecutor.executeWithParentIO(commands);
+			}
+			break;
+
+		default:
+			break;
```

Comment:
In default case should we throw exception saying K8SAuthorisation not yet supported?